source/code/plugins/oms_configuration.rb

```diff
@@ -80,17 +81,42 @@ def parse_proxy_config(proxy_conf_str)
           # Convert nammed matches to a hash
           Hash[ matches.names.map{ |name| name.to_sym}.zip( matches.captures ) ]
       end
+      
+      def get_azure_region_from_imds()
+          begin
+            uri = URI.parse(@@AzureIMDSEndpoint)
+            http_get_req = Net::HTTP::Get.new(uri, initheader = {'Metadata' => 'true'})
+
+            http_req = Net::HTTP.new(uri.host, uri.port)
+
+            http_req.open_timeout = 3
+            http_req.read_timeout = 2
+
+            res = http_req.start() do |http|
+              http.request(http_get_req)
+            end
+
+            imds_instance_json = JSON.parse(res.body)
+
+            return nil if !imds_instance_json.has_key?("compute") || imds_instance_json['compute'].empty? #classic vm
 
+            imds_instance_json_compute = imds_instance_json['compute']
+            return nil unless imds_instance_json_compute.has_key?("location")
+            return nil if imds_instance_json_compute['location'].empty?
+            return imds_instance_json_compute['location']
+          rescue => e
+            # this may be a container instance or a non-Azure VM
+            OMS::Log.warn_once("Could not fetch Azure Resource ID from IMDS, Reason: #{e}")
+            return nil
+          end
+      end  
+      
       def get_azure_resid_from_imds()
           begin
             uri = URI.parse(@@AzureIMDSEndpoint)
             http_get_req = Net::HTTP::Get.new(uri, initheader = {'Metadata' => 'true'})
 
-            if @@ProxyConfig.empty?
-              http_req = Net::HTTP.new(uri.host, uri.port)
-            else
-              http_req = Net::HTTP.new(uri.host, uri.port, @@ProxyConfig[:addr], @@ProxyConfig[:port], @@ProxyConfig[:user], @@ProxyConfig[:pass])
```

Comment:
if proxy shouldn't beused for IMDS, why was this specific case added initially? was there any historical reason?